BrainCloudClient/Assets/BrainCloud/Client/BrainCloud/BrainCloudFile.cs

```diff
@@ -102,10 +109,9 @@ public BrainCloudFile(BrainCloudClient client)
             data[OperationParam.UploadCloudPath.Value] = cloudPath;
             data[OperationParam.UploadShareable.Value] = shareable;
             data[OperationParam.UploadReplaceIfExists.Value] = replaceIfExists;
-
-            byte[] file = System.Convert.FromBase64String(fileData);
-            data[OperationParam.UploadFileSize.Value] = file.Length;
-
+            
+            data[OperationParam.UploadFileSize.Value] = FileStorage[fileData].Length;
```

Comment:
This seems wrong.
you haven't add fileData to storage yet. Also wouldn't the key be your GUID?